nixos/modules/services/security/esdm.nix

```diff
@@ -0,0 +1,138 @@
+  linux_6_3 = (lib.versionAtLeast kernelVersion "6.3");
+in
+{
+  options.services.esdm = {
+    enable = lib.mkEnableOption (lib.mdDoc "ESDM service configuration");
+    package = lib.mkPackageOptionMD pkgs "esdm" { };
+    kernelSupportEnable = lib.mkEnableOption (lib.mdDoc "Enable kernel support for the scheduler- and interrupt-entropy-source.");
+    serverEnable = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = lib.mdDoc ''
+        Enable option for ESDM server service. If serverEnable == false, then the esdm-server
+        will not start. Also the subsequent services esdm-cuse-random, esdm-cuse-urandom
+        and esdm-proc will not start as these have the entry Want=esdm-server.service.
+      '';
+    };
+    cuseRandomEnable = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = lib.mdDoc ''
+        Enable option for ESDM cuse-random service. Determines if the esdm-cuse-random.service
+        is started.
+      '';
+    };
+    cuseUrandomEnable = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = lib.mdDoc ''
+        Enable option for ESDM cuse-urandom service. Determines if the esdm-cuse-urandom.service
+        is started.
+      '';
+    };
+    procEnable = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = lib.mdDoc ''
+        Enable option for ESDM proc service. Determines if the esdm-proc.service
+        is started.
+      '';
+    };
+    verbose = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Enable verbose ExecStart for ESDM. If verbose == true, then the corresponding "ExecStart"
+        values of the 4 aforementioned services are overwritten with the option
+        for the highest verbosity.
+      '';
+    };
+    esdmHashName = lib.mkOption {
+      type = lib.types.str;
+      default = "sha3-512";
+      description = lib.mdDoc ''
+        Set the hash configration of the kernel module esdm_es.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable (
+    lib.mkMerge [
+      ({
+        systemd.packages = [ cfg.package ];
+      })
+      (lib.mkIf cfg.kernelSupportEnable {
+        assertions = lib.lists.singleton {
+          assertion = linux_6_3;
+          message = "ESDM expects kernel version >= 6.3. got kernelversion:${kernelVersion}";
+        };
+
+        boot.extraModulePackages = [ config.boot.kernelPackages.esdm_es.out ];
+        boot.extraModprobeConfig = ''
+          options esdm_es esdm_hash_name=${cfg.esdmHashName}
+        '';
+        boot.kernelModules = [ "esdm_es" ];
+
+        #patch kernel (works for kernel version 6.3)
+        boot.kernelPatches = [ ]
+          ++ lib.lists.optionals linux_6_3 [
+          {
+            name = "esdm_sched_es_hook";
+            patch = "${cfg.package}/share/linux_esdm_es/0001-ESDM-scheduler-entropy-source-hooks_6.4.patch";
```

Comment:
Those are 6.4 only patches right? Why not exact match over 6.4 rather than saying >= 6.3 which will fail with our 6.5 RCs?

Reply:
The patches work with both 6.3 and 6.4. Future possible incompatibilities would be addressed when then show up, if that is okay.

Reply:
That's totally okay, please take a look to latest master branch, we have 6.4 now, so you can test it too. :)